boundlessgeo/exchange
# Open bex 1007

## JIRA Ticket
[BEX-1007](https://issues.boundlessgeo.com:8443/browse/BEX-1007)

## Description
This is bringing over the PR from the previous exchange repo into the current one, as it was not previously merged.

Adds the `Content Manager` and `Content Creator` fields via the `ExchangeProfile` model. The "Content Manager" field determines if a user may register remote services. The "Content Creator" field determines if a user may upload documents or layers. `ExchangeProfile` has a one-to-one relationship to the `Profile` model from GeoNode, effectively extending it with these extra fields.

The templates and views have been updated to use these fields to prevent operations as necessary. The user is allowed to go to the view, but the buttons will be disabled, and `POST` requests ignored if the user does not have the correct permissions.

## TODO

## Steps to Test or Reproduce
Exchange setup:
Run migrations. `docker-compose exec exchange /bin/bash` to enter the Exchange instance, then `python manage.py migrate` to run migrations.

Exchange testing:
1. Log in as `admin`
2. Navigate to the Django admin panel > Exchange Core > Exchange Profiles
3. Click on `test` to modify the `test` user. Deselect `Content Creator` and `Content Manager` and hit `Save`.
4. Log out of `admin` and log in to `test`.

Finally, try the following, and you should always be prevented via disabled buttons:
1. Data > Manage Remote Services
2. Manually navigate to `/services/register/` in the URL
3. Data > Create Layer
4. Data > Upload Layer
5. Data > Upload Document

## Setup Environment with PR

1. Cleanup previous state

```bash
make purge
```

2. Checkout PR

__Using git command (command line interface)__
```bash
pr_id=ADD_PR_NUMBER_HERE
git checkout master
git branch -D $pr_id
git fetch
git fetch origin pull/$pr_id/head:$pr_id
git checkout $pr_id
git submodule init
git submodule update --remote --recursive
```

__Using [Gitkraken](https://www.gitkraken.com/)__

+ In GitKraken, right click on the pull request you want to review. Select Add Remote and Checkout (the PR).

3. Start exchange

```bash
make start
```

4. Exchange Healthcheck

```bash
docker inspect --format '{{ .State.Health.Status }}' exchange
```

__NOTE:__ Only continue the following steps if the output from the above command is `healthy`. You may have to wait 
a few minutes.

---
@boundlessgeo/bex-qa
